[python] Creation of a DICOM archive validation script

cmadjar opened the pull request:
### Description

This is the first official PR of many to implement an equivalent of `tarchiveLoader.pl` on the python's side which will be used to create BIDS datasets out of DICOM archives.

This script does the following validation on a DICOM archive:
- Verification of the DICOM archive against the checksum stored in the database
- Verification of the PSC information using either PatientName or PatientID DICOM header
- Verification of the ScannerID in the database (optionally create a new one if necessary)
- Validity check of the candidate IDs associated to the DICOM archive based on the PatientName
- Validity check of the session associated to the DICOM archive based on the PatientName
- Update to the mri_upload's isTarchiveValidated field if above validations were successful

### Notes for existing project planning on using the python set of scripts

Ensure to update you database_config.py with the following content:
```
import re
from lib.imaging import Imaging

def get_subject_ids(db, dicom_value=None, scanner_id=None):

    subject_id_dict = {}

    imaging = Imaging(db, False)

    phantom_match   = re.search('(pha)|(test)', dicom_value, re.IGNORECASE)
    candidate_match = re.search('([^_]+)_(\d+)_([^_]+)', dicom_value, re.IGNORECASE)

    if phantom_match:
        subject_id_dict['isPhantom']  = True
        subject_id_dict['CandID']     = imaging.get_scanner_candid(scanner_id)
        subject_id_dict['visitLabel'] = dicom_value.strip()
        subject_id_dict['createVisitLabel'] = 1
    elif candidate_match:
        subject_id_dict['isPhantom']  = False
        subject_id_dict['PSCID']      = candidate_match.group(1)
        subject_id_dict['CandID']     = candidate_match.group(2)
        subject_id_dict['visitLabel'] = candidate_match.group(3)
        subject_id_dict['createVisitLabel'] = 0

    return subject_id_dict
```

cmadjar commented:
Note to self, need to replace the `-nonewscanner` script option by the new config option created in this PR on the LORIS side: https://github.com/aces/Loris/pull/5954

jesscall commented:
update instructions on how to update `database_config.py` file

jesscall commented on dicom-archive/database_config_template.py:
```diff
@@ -6,4 +6,27 @@
+
+    subject_id_dict = {}
+
+    imaging = Imaging(db, False)
```
modify to add the import otherwise get the following error:
```
Traceback (most recent call last):
  File "python/dicom_archive_validation.py", line 316, in <module>
    main()
  File "python/dicom_archive_validation.py", line 82, in main
    validate_dicom_archive(config_file, tarchive_path, upload_id, verbose)
  File "python/dicom_archive_validation.py", line 258, in validate_dicom_archive
    site_dict = imaging_obj.determine_study_center(tarchive_info_dict)
  File "/data/ccna/bin/mri-test/python/lib/imaging.py", line 434, in determine_study_center
    subject_id_dict = self.determine_subject_ids(tarchive_info_dict)
  File "/data/ccna/bin/mri-test/python/lib/imaging.py", line 351, in determine_subject_ids
    subject_id_dict = self.config_file.get_subject_ids(self.db, dicom_value, scanner_id)
  File "/data/ccna/bin/mri-test/dicom-archive/.loris_mri/database_config.py", line 15, in get_subject_ids
    imaging = Imaging(db, False)
NameError: name 'Imaging' is not defined
```

jesscall replied:
+ re import

cmadjar commented:
@jesscall I updated the database_config_template.py file and updated the description of the PR to tell the users what to update.

Let me know if you see anything else. Thank you!!

jesscall commented:
looks good!
